model.py

```diff
@@ -0,0 +1,177 @@
+import numpy as np
+import tensorflow as tf
+
+from neuraxle.hyperparams.space import HyperparameterSamples
+from neuraxle.pipeline import MiniBatchSequentialPipeline, Pipeline, Joiner
+from neuraxle.steps.data import EpochRepeater
+from neuraxle_tensorflow.tensorflow_v1 import TensorflowV1ModelStep
+
+from datasets import generate_x_y_data_v1, generate_x_y_data_v2, generate_x_y_data_v3, generate_x_y_data_v4
+
+GO_TOKEN = -1.
+
+
+def create_graph(step: TensorflowV1ModelStep):
+    # shape: (batch_size, seq_length, input_dim)
+    data_inputs = tf.placeholder(
+        dtype=tf.float32,
+        shape=[None, None, step.hyperparams['input_dim']],
+        name='data_inputs'
+    )
+
+    # shape: (batch_size, seq_length, input_dim)
+    expected_outputs = tf.placeholder(
+        dtype=tf.float32,
+        shape=[None, None, step.hyperparams['output_dim']],
+        name='expected_outputs'
+    )
+
+    # shape: (batch_size)
+    target_sequence_length = tf.placeholder(dtype=tf.int32, shape=[None], name='expected_outputs_length')
+
+    encoder_state = create_encoder(step, data_inputs)
+
+    decoder_cell = create_stacked_rnn(step)
+    decoder_outputs_training = create_training_decoder(step, encoder_state, decoder_cell)
+    decoder_outputs_inference = create_inference_decoder(step, encoder_state, decoder_cell)
+
+    return decoder_outputs_training, decoder_outputs_inference
+
+
+def create_encoder(step, data_inputs):
+    encoder_cell = create_stacked_rnn(step)
+    encoder_outputs, encoder_state = tf.nn.dynamic_rnn(cell=encoder_cell, inputs=data_inputs, dtype=tf.float32)
+
+    return encoder_state
+
+
+def create_training_decoder(step: TensorflowV1ModelStep, encoder_state, decoder_cell):
+    go_tokens = tf.constant(GO_TOKEN, shape=[step.hyperparams['batch_size'], 1, step.hyperparams['output_dim']])
+    inputs = tf.concat([go_tokens, step['expected_outputs']], axis=1)
+
+    helper = tf.contrib.seq2seq.TrainingHelper(
+        inputs=inputs,
+        sequence_length=step['expected_outputs_length']
+    )
+
+    output = create_decoder_outputs(step, helper, encoder_state, decoder_cell)
+
+    return output
+
+
+def create_inference_decoder(step: TensorflowV1ModelStep, encoder_state, decoder_cell):
+    start_inputs = tf.constant(GO_TOKEN, shape=[step.hyperparams['batch_size'], step.hyperparams['output_dim']])
+
+    helper = tf.contrib.seq2seq.InferenceHelper(
```

Comment:
I am stuck here with the InferenceHelper.... At least, the TrainingHelper is training well ! Any ideas why the InferenceHelper is not working ? @guillaume-chevalier

Reply:
it works now.

Reply:
fixed with maximum iterations parameter